server/lib/src/constants/acp.rs

```diff
@@ -17,6 +17,24 @@ pub struct BuiltinAcp {
     receiver_group: Uuid,
     target_scope: ProtoFilter,
     search_attrs: Vec<Attribute>,
+    modify_removed_attrs: Vec<Attribute>,
+    modify_classes: Vec<EntryClass>,
+}
+
+impl Default for BuiltinAcp {
+    fn default() -> Self {
+        Self {
+            classes: Default::default(),
+            name: Default::default(),
+            uuid: Default::default(),
+            description: Default::default(),
+            receiver_group: Default::default(),
+            search_attrs: Default::default(),
+            modify_removed_attrs: Default::default(),
+            modify_classes: Default::default(),
+            target_scope: ProtoFilter::SelfUuid,
```

Comment:
Not sure how I feel about this. Maybe it should be an empty filter instead so nothing matches?

Reply:
Ah, I *knew* there were some things I'd forgotten to check with you on :D

How do we define an "empty" filter in this context?

Reply:
Try `And(vec::with_capacity(0))`. An And filter with no elements should never match anything per entry_match_no_index_inner in server/lib/src/entry.rs. Specifically if you have a rust iter with .all() and no elements contained then it returns false https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any

Reply:
![image](https://github.com/kanidm/kanidm/assets/271005/dae50306-4050-4b50-8274-72db8051a109)

I was there when the deep magic was written.